rust/crd/src/lib.rs

```diff
@@ -287,24 +328,29 @@ impl SparkApplication {
             format!("--conf spark.kubernetes.driver.container.image={}", self.spec.spark_image.as_ref().context(NoSparkImageSnafu)?),
             format!("--conf spark.kubernetes.executor.container.image={}", self.spec.spark_image.as_ref().context(NoSparkImageSnafu)?),
             format!("--conf spark.kubernetes.authenticate.driver.serviceAccountName={}", serviceaccount_name),
-        ];
+        ]);
 
         // See https://spark.apache.org/docs/latest/running-on-kubernetes.html#dependency-management
         // for possible S3 related properties
         if let Some(endpoint) = s3bucket.as_ref().and_then(|s3| s3.endpoint()) {
             submit_cmd.push(format!("--conf spark.hadoop.fs.s3a.endpoint={}", endpoint));
         }
-        if s3bucket.as_ref().and_then(|s3| s3.secret_class()).is_some() {
-            // We don't use the secret at all here, instead we assume the Self::env() has been
-            // called and this environment variables are availables.
-            submit_cmd.push(format!(
-                "--conf spark.hadoop.fs.s3a.access.key=${}",
-                ENV_AWS_ACCESS_KEY_ID
-            ));
-            submit_cmd.push(format!(
-                "--conf spark.hadoop.fs.s3a.secret.key=${}",
-                ENV_AWS_SECRET_ACCESS_KEY
-            ));
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            if let Some(S3AccessStyle::Path) = conn.access_style {
```

Comment:
@teozkr and I prefer having a `match conn.access_style {...}` statement here to get a compilation error when a new accessStyle is introduced rather than silently not supporting it

Reply:
Nice point: I'll change that.

Reply:
Changed.